### PR TITLE
Adding a fix for the no certs.

### PR DESCRIPTION
**Pull request by THOR300**
# Description

This pull request integrates a new sdk version and new method of instantiation of the search adaptor object. This can be utilised across the other failing tests. 

The issue: 
We were using the following code to instantiate the `search._VESPA_CONNECTION` object, we were therefore passing in cert paths that didn't exist that requests then couldn't find. Now we allow no cert paths in the sdk. 

```python
_VESPA_CONNECTION = VespaSearchAdapter(
    instance_url=VESPA_URL,
    cert_directory=VESPA_SECRETS_LOCATION,
    embedder=ENCODER,
)
```

- [ ] TODO: Update the lock file when poetry and pypi decide to behave and identify the update. 

## Proposed version

Please select the option below that is most relevant from the list below. This
will be used to generate the next tag version name during auto-tagging.

- [ ] Skip auto-tagging
- [x] Patch
- [ ] Minor version
- [ ] Major version

Visit the [Semver website](https://semver.org/#summary) to understand the
difference between `MAJOR`, `MINOR`, and `PATCH` versions.

Notes:

- If none of these options are selected, auto-tagging will fail
- Where multiple options are selected, the most senior option ticked will be
  used -- e.g. Major > Minor > Patch
- If you are selecting the version in the list above using the textbox, make
  sure your selected option is marked `[x]` with no spaces in between the
  brackets and the `x`

## Type of change

Please select the option(s) below that are most relevant:

- [x] Bug fix
- [ ] New feature
- [ ] Breaking change
- [ ] GitHub workflow update
- [ ] Documentation update
- [ ] Refactor legacy code
- [ ] Dependency update

## How Has This Been Tested?

Please describe the tests that you added to verify your changes.

## Reviewer Checklist

- [ ] **DB_CLIENT DEPENDENCY IS ON THE LATEST VERSION**
- [ ] The PR represents a single feature (small driveby fixes are also ok)
- [ ] The PR includes tests that are sufficient for the level of risk
- [ ] The code is sufficiently commented, particularly in hard-to-understand areas
- [ ] Any required documentation updates have been made
- [ ] Any TODOs added are captured in future tickets
- [ ] No FIXMEs remain

### CODE REVIEW

File: tests/search/test_vespasearch.py [hunk 1]
```diff
@@ -853,7 +856,12 @@ def test_csv_download_search_variable_limit(
     # monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
     _populate_db_families(data_db)
 
-    query_spy = mocker.spy(search._VESPA_CONNECTION, "search")
```
Comment (katybaulch) [inline tests/search/test_vespasearch.py:856]: There are two instances of this mocker.spy in this file - do we need to do this for the other line too?

**Review by katybaulch — APPROVED** [review]
lgtm once poetry's updated - thanks for doing this 🙏🏻

File: tests/search/test_vespasearch.py [hunk 2]
```diff
@@ -853,7 +856,12 @@ def test_csv_download_search_variable_limit(
     # monkeypatch.setattr(search, "_VESPA_CONNECTION", test_vespa)
```
Comment (katybaulch) [inline tests/search/test_vespasearch.py:856]: Commented out code.

File: tests/search/test_vespasearch.py (hunk 1 above)
Comment (THOR300) [reply tests/search/test_vespasearch.py:856]: Yep!

**Pull request closed without merging**